### PR TITLE
std: add cross platform null-device name

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description
We have meet a discord discussion about cross platform `null-device`: https://discord.com/channels/601130461678272522/988303282931912704/1165966467095875624

I want the same feature too...And I think it's good enough to add it into `nu-std`.

Different to https://github.com/nushell/nu_scripts/pull/649/files, I think named it to `null-device`(the name comes from [wiki](https://en.wikipedia.org/wiki/Null_device)) is better than `null-stream`.

### CODE REVIEW

File: crates/nu-std/std/mod.nu [hunk 1]
```diff
@@ -335,3 +335,16 @@ export def repeat [
 
     1..$n | each { $item }
 }
+
+# return a null device file.
+#
+# # Examples
+#     run a command and ignore it's stderr output
+#     > cat xxx.txt e> (null-device)
+export def null-device []: any -> string {
```
Comment (amtoine) [inline crates/nu-std/std/mod.nu:344]: should it take `any`?

File: crates/nu-std/std/mod.nu [hunk 2]
```diff
@@ -335,3 +335,16 @@ export def repeat [
...
+    if ((sys).host.name | str downcase) == "windows" {
```
Comment (amtoine) [inline crates/nu-std/std/mod.nu:345]: i find the `$nu.os-info.name` version easier to read but ehe :wink:

File: crates/nu-std/std/mod.nu [hunk 3]
```diff
@@ -335,3 +335,16 @@ export def repeat [
...
+        "NUL"
```
Comment (amtoine) [inline crates/nu-std/std/mod.nu:346]: should that be `NUL:` as in the [stdlib candidate addition](https://github.com/nushell/nu_scripts/pull/649/files#diff-92aa723c230c96a675a0e1deacfb29994e5efce3d6909713d7c0f3d9a84681d5R6)?
i thought thre was a reason for that

**Review by amtoine — COMMENTED** [review]
once we land this new command, could you remove the `null-stream` command from the `nu_scripts` @WindSoilder ? :innocent:

File: crates/nu-std/std/mod.nu (hunk 3 above)
Comment (ChrisDenton) [reply crates/nu-std/std/mod.nu:346]: Not to bikeshed too much but I think it'd be better to use the full path `\\.\NUL`. While a plain `NUL` *should* normally get redirected to the proper path, I wouldn't want to rely on that in all contexts.

Comment (amtoine) [reply crates/nu-std/std/mod.nu:346]: good to know :pray: 

then the annotation might become `-> path` to reflect the change :+1:

File: crates/nu-std/std/mod.nu (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-std/std/mod.nu:344]: Actually I'm not really sure, it doesn't deal with any pipeline input.  Maybe changed to nothing?

File: crates/nu-std/std/mod.nu (hunk 3 above)
Comment (WindSoilder) [reply crates/nu-std/std/mod.nu:346]: Thanks!  It's good to know the `\\.\NUL`, and I verified it works on my windows machine

File: crates/nu-std/std/mod.nu (hunk 2 above)
Comment (WindSoilder) [reply crates/nu-std/std/mod.nu]: Thanks!  And the output of `$nu.os-info.name` seems more stable

Comment (amtoine) [reply crates/nu-std/std/mod.nu]: yup :yum:

Comment (amtoine) [reply crates/nu-std/std/mod.nu]: not even sure we need to `str downcase` it but that's fine :wink:

**Review by amtoine — APPROVED** [review]
thanks @WindSoilder for this and tweaking the code from review comments :pray:

**Comment by Hullabaloo-and-Howdy-Do** [comment]
Redirecting to the virtual path that Windows uses does the job, but I think the proper way of doing it is actually to redirect to `NUL:`

**Comment by ChrisDenton** [comment]
`\\.\NUL` is the canonical win32 path, try using [`GetFullPathName`](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfullpathnamew) on `NUL`. Things like `NUL:` are an old bit of wisdom from the days before UNC paths, device paths and file streams when it was used as an illegal file name that should always error if not redirected and thus shouldn't conflict with any real file. That's a more fragile assumption.

**Comment by Hullabaloo-and-Howdy-Do** [comment]
> `\\.\NUL` is the canonical win32 path, try using [`GetFullPathName`](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfullpathnamew) on `NUL`. Things like `NUL:` are an old bit of wisdom from the days before UNC paths, device paths and file streams when it was used as an illegal file name that should always error if not redirected and thus shouldn't conflict with any real file. That's a more fragile assumption.

Ahh, interesting. I wasn't aware of that, I had trouble finding any documentation on the null device and the `NUL:` thing was the only detail I could find. Good to know.

**Pull request merged**